### PR TITLE
Add IntervalSequence,

### PR DESCRIPTION
**Pull request by matthewmeyer**
Add an add function to IntervalSequence

### CODE REVIEW

File: src/intervalSequence.ts [hunk 1]
```diff
@@ -129,3 +136,31 @@ export const merge
...
+
+export type Add
+    =  <E, A>(_: Strategy<E, A>)
+    => (_: IntervalSequence<E, A>)
+    => (_: Interval<E, A | undefined>)
+    => IntervalSequence<E, A>
+
+export const add
+    : Add
+    = intervalSequence => current => toAdd => {
```
Comment (sergey-shandar) [inline src/intervalSequence.ts]:
```suggestion
    = strategy => current => toAdd => {
```

File: src/intervalSequence.ts [hunk 2]
```diff
@@ -129,3 +136,31 @@ export const merge
...
+            first: undefined,
+            rest: sequence.fromArray([
+                { edge: toAdd.min, value: toAdd.value },
+                { edge: toAdd.max, value: undefined }
+            ])
+        }
+
+        return merge<E, A, A | undefined, A>({
+            intervalSequence,
+            reduce: a => b => b === undefined ? a : b
```
Comment (sergey-shandar) [inline src/intervalSequence.ts]: could we make it as a static function (not exported)? The reason is that this function doesn't require context so less stackframes.

File: src/intervalSequence.ts [hunk 3]
```diff
@@ -129,3 +136,31 @@ export const merge
...
+
+        const mergeSeq: IntervalSequence<E, A | undefined> = {
+            first: undefined,
+            rest: sequence.fromArray([
+                { edge: toAdd.min, value: toAdd.value },
+                { edge: toAdd.max, value: undefined }
+            ])
+        }
+
+        return merge<E, A, A | undefined, A>({
```
Comment (sergey-shandar) [inline src/intervalSequence.ts]: please, avoid type annotations in function calls. It's not compatible with current fun.ts language. use
```
const NAME
    : TYPE_DEFINITION
   = VALUE
return NAME
```

**Comment by sergey-shandar** [comment]
Few minor comments, otherwise LGTM

File: src/intervalSequence.ts (hunk 2 above)
Comment (matthewmeyer) [reply src/intervalSequence.ts]: refactored the merge method signature in most recent version

File: src/intervalSequence.ts (hunk 3 above)
Comment (matthewmeyer) [reply src/intervalSequence.ts]: fixed

File: src/intervalSequence.ts [hunk 4]
```diff
@@ -42,7 +44,8 @@ const drop
 export type Reduce<A, B, R> = (_: A) => (_: B) => R
 
 export type Merge
-    = <E, A, B, R>(_: MergeStrategy<E, A, B, R>)
+    = <E, A, B, R>(_: Strategy<E, R>)
+    => (_: Reduce<A, B, R>)
```
Comment (sergey-shandar) [inline src/intervalSequence.ts]:
```suggestion
    => <A, B>(_: Reduce<R>)
```

File: src/intervalSequence.ts [hunk 5]
```diff
@@ -42,7 +44,8 @@ const drop
 export type Reduce<A, B, R> = (_: A) => (_: B) => R
 
 export type Merge
-    = <E, A, B, R>(_: MergeStrategy<E, A, B, R>)
+    = <E, A, B, R>(_: Strategy<E, R>)
```
Comment (sergey-shandar) [inline src/intervalSequence.ts]:
```suggestion
    = <E, R>(_: Strategy<E, R>)
```

File: src/intervalSequence.ts [hunk 6]
```diff
@@ -129,3 +132,34 @@ export const merge
...
+                { edge: toAdd.min, value: toAdd.value },
+                { edge: toAdd.max, value: undefined }
+            ])
+        }
+
+        const reduce
+            : (_: A) => (_: A | undefined) => A
+            = a => b => b === undefined ? a : b
+
+        const returnValue = merge<E, A, A | undefined, A>(strategy)(reduce)(current)(mergeSeq)
```
Comment (sergey-shandar) [inline src/intervalSequence.ts]:
```suggestion
        const returnValue
             : Merge<E, A, A | undefined, A>
             = merge(strategy)(reduce)(current)(mergeSeq)
```

**Review by sergey-shandar — COMMENTED** [review]
`package.json` version

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**